server/app/service/package.scala

```diff
@@ -1,23 +1,43 @@
-import shared.model.{TestResult, TestStatus}
+import monix.execution.cancelables.BooleanCancelable
+import shared.model.{Event, TestResult, TestStatus}
+import shared.view.SuiteReportUtil.red
 
 import scala.util.{Failure, Success, Try}
 
 package object service {
   //scalatest keyword
   val testFailedMarker = "FAILED"
 
-  implicit class RichRunnerString(val output: Try[String]) extends AnyVal {
-    def check = output.filter(!_.contains(testFailedMarker))
+  type CheckNext = (String) => Unit
+
+  type PushResult = (Event) => Unit
+
+  type OnBlockComplete = Try[Unit] => Unit
+
+  def testSync: (CheckNext, (=> Unit) => Event) = {
```

Comment:
Is it possible to name this horrible type as well -  (=> Unit) => Event ? Maybe it will add more understanding when reading this code